cellmlmanip/model.py

```diff
@@ -1041,6 +1043,34 @@ def _convert_variable_instance(self, original_variable, cf, units, direction, mo
 
         return new_variable
 
+    def remove_fixable_singularities(self, V, exclude=set()):
+        """
+        Removes removable singularities from the model equations and replaces these with a piecewise.
+
+        The process looks for equations of any of the following forms, where U is a function of V:
+        - `U / (exp(U) - 1.0)`
+        - `U / (1.0 - exp(U))`
+        - `(exp(U) - 1.0) / U`
+        - `(1.0 - exp(U)) / U`
+        It replaces these with a piecewise 1e-7 either side of U==0 drawing a stright line in the region.
+        For example `(V + 5)/(exp(V + 5) - 1)` becomes
+        `((fabs(-V - 5.0000000000000000) < fabs(-4.9999999000000000 / 2 - -5.0000001000000000 / 2))
+        ? -0.494049243462503*V - 1.4702462167574 : ((5.0 + V) / (-1.0 + exp(5.0 + V))))`
+
+        See for more details appendix B in:
+        [Johnstone, R. H. (2018). Uncertainty characterisation in action potential modelling for cardiac drug safety.
```

Comment:
This isn't RST syntax!

Reply:
is it?

Reply:
It's hard to find exactly what the right forwmat is but I think it's ok the way it's now (i'm making a small tweak).